primitives/bignumber.go

```diff
@@ -0,0 +1,560 @@
+package primitives
+
+import (
+	"encoding/hex"
+	"github.com/ethereum/go-ethereum/rlp"
+	"math/big"
+)
+
+type U64 big.Int
+type U128 big.Int
+type U256 big.Int
+
+func (u U64) String() string {
+	var a = big.Int(u)
+	return a.String()
+}
+
+func NewU64(v string) U64 {
+	var x big.Int
+	x.SetString(v, 10)
+	return U64(x)
+}
+
+func (z *U64) set(y big.Int) {
+	*z = U64(y)
+}
+
+func (z *U64) Add(x *U64, y *U64) *U64 {
```

Comment:
Wow Golang has very interesting convention.

Reply:
https://golang.org/pkg/math/big/#Int.Add
I don't know whether this is a widely used Golang convention or not.

But I think comments(function descriptions) should be added since this is a very interesting convention.

Reply:
I think it is natural in Golang.
Their types have an initial zero value each.
In the `sum` case, the result value is initialized as zero value and the sum value will be written by `sum` function.
Also by using this convention, they can remove unintentional memory allocation. The `Add` function does not create any garbage.